test_prime_factors.py

```diff
@@ -0,0 +1,117 @@
+from typing import Any, List
+import pytest
+from prime_factors import prime_factors
+
+
+def test_prime_factors_returns_list() -> None:
+    result = prime_factors(number=2)
+    assert isinstance(result, list), \
+        f"Expected result to be a list, got {type(result)}!"
+
+
+@pytest.mark.parametrize(
+    "data,expectation",
+    [
+        (2.0, "Expected integer, got <class 'float'>!"),
+        ("2.0", "Expected integer, got <class 'str'>!"),
+        ([], "Expected integer, got <class 'list'>!"),
+    ],
+)
+def test_prime_factors_raises_valueerror_when_number_is_not_an_integer(data: Any, expectation: str) -> None:
+    with pytest.raises(ValueError) as value_error:
+        prime_factors(number=data)
+    assert str(value_error.value) == expectation, \
+        f"Expected '{expectation}', got '{str(value_error.value)}' for '{data}'!"
```

Comment:
Nie wystarcza domyślny poziom raportowania pytesta?

```
E       assert "Expected int...lass 'list'>!" == "Expected int...lass 'list'>!"
E         - Expected integer, got <class 'list'>!
E         + Expected inteeeeger, got <class 'list'>!
E         ?             +++
```

Reply:
Zgadzam się, przerobiłem.
W pierwszej kolejności zrobiłem to idąc za przykładami z materiałów o tic_tac_toe, lecz racja tam nie był pytest wykorzystywany.